internal/engine/upper.go

```diff
@@ -400,6 +405,29 @@ func handleStartProfilingAction(state *store.EngineState) {
 	state.IsProfiling = true
 }
 
+func handleHeapProfileAction(ctx context.Context) {
```

Comment:
i don't understand why this is part of the engine state loop at all? it doesn't actually read any engine state

Reply:
(seems like it'd be much simpler as a library function we call directly from the HUD)

Reply:
Hm, yeah, my mental model was this is the "respond to action" loop rather than the "engine state" loop.
It ended up here because: the most cohesive place (from the perspective of this feature, but rather uncohesive from the perspective of hud.go) to put it was in the `case ^O:` right in hud.go. I perhaps prematurely generalized and figured we might want to allow that from the web ui, and making it happen in response to an action seemed like a pretty good way to allow that.

Apparently at some point I started thinking of actions as "make a thing happen" rather than a mechanism for serializing state mutation.